atomicapp/applogging.py

```diff
@@ -0,0 +1,153 @@
+        return "\033[" + colorCodes[color] + "m" + text + "\033[0m"
+
+    def _make_unicode(self, input):
+        """
+        Convert all input to utf-8 for multi language support
+        """
+        if type(input) != unicode:
+            input = input.decode('utf-8')
+        return input
+
+
+class Logging:
+
+    @staticmethod
+    def setup_logging(verbose=None, quiet=None, logtype=None):
+        """
+        This function sets up logging based on the logtype requested.
+        The 'none' level outputs no logs at all
+        The 'cockpit' level outputs just logs for the cockpit logger
+        The 'nocolor' level prints out normal log msgs (no cockpit) without color
+        The 'color' level prints out normal log msgs (no cockpit) with color
+        """
+
+        # If no logtype was set then let's have a sane default
+        # If connected to a tty, then default to color, else, no color
+        if not logtype:
+            if sys.stdout.isatty():
+                logtype = 'stdoutcolor'
+            else:
+                logtype = 'stdout'
+
+        # Determine what logging level we should use
+        if verbose:
+            logging_level = logging.DEBUG
+        elif quiet:
+            logging_level = logging.WARNING
+        else:
+            logging_level = logging.INFO
+
+        # Get the loggers and clear out the handlers (allows this function
+        # to be run more than once)
+        logger = logging.getLogger(LOGGER_DEFAULT)
+        logger.handlers = []
+        cockpit_logger = logging.getLogger(LOGGER_COCKPIT)
+        cockpit_logger.handlers = []
+
+        if logtype == 'none':
+            # blank out both loggers
+            logger.addHandler(logging.NullHandler())
+            cockpit_logger.addHandler(logging.NullHandler())
+            return
+
+        if logtype == 'cockpit':
+            # blank out normal log messages
+            logger.addHandler(logging.NullHandler())
+
+            # configure cockpit logger
+            handler = logging.StreamHandler(stream=sys.stdout)
+            formatter = logging.Formatter('atomicapp.status.%(levelname)s.message=%(message)s')
+            handler.setFormatter(formatter)
+            cockpit_logger.addHandler(handler)
+            cockpit_logger.setLevel(logging_level)
+            return
+
+        if logtype == 'stdout':
+            # blank out cockpit log messages
+            cockpit_logger = logging.getLogger(LOGGER_COCKPIT)
+            cockpit_logger.addHandler(logging.NullHandler())
+
+            # configure logger for basic printing to stdout
+            handler = logging.StreamHandler(stream=sys.stdout)
+            formatter = logging.Formatter('%(asctime)s - [%(levelname)s] - %(filename)s - %(message)s')
+            handler.setFormatter(formatter)
+            logger.addHandler(handler)
+            logger.setLevel(logging_level)
+            return
+
+        if logtype == 'stdoutcolor':
+            # blank out cockpit log messages
```

Comment:
no, it gets created befores these if statements. It was a mistake to have `cockpit_logger = logging.getLogger(LOGGER_COCKPIT)` in the if statement above. removing it now.